### PR TITLE
add string mixin to get the index of a matching regex

### PR DESCRIPTION
**Pull request by grounded042**
This will solve https://github.com/gluck/il-repack/issues/154 by matching either `, PublicKey=` or `,PublicKey=`.

### CODE REVIEW

File: ILRepack/Steps/AttributesRepackStep.cs [hunk 1]
```diff
@@ -88,7 +90,7 @@ void CleanupAttributes()
             {
                 String name = (string)ca.ConstructorArguments[0].Value;
                 int idx;
-                if ((idx = name.IndexOf(", PublicKey=", StringComparison.Ordinal)) != -1)
+                if ((idx = name.IndexOfRegex(new Regex(", ?PublicKey="))) != -1)
```
Comment (gluck) [inline ILRepack/Steps/AttributesRepackStep.cs]: while it perfectly does the trick here, the allowed format is more permissive, e.g. you can have quotes around the value, or spaces around the '=', and the 'publickey' case doesn't matter.
I thought using `new AssemblyName(str).GetPublicKeyToken() != null` would work just fine w/o us having to care about the format details, but it doesn't like `PublicKey=null` which seems to be valid in `InternalsVisibleTo` references... :sob:

Comment (timotei) [inline ILRepack/Steps/AttributesRepackStep.cs]: Isn't it fine to take the name from 0 to first occurrence  + trim? Then we don't care about the publickey part at all

Comment (gluck) [inline ILRepack/Steps/AttributesRepackStep.cs]: indeed (but won't work for the below one)

Comment (AColmant) [inline ILRepack/Steps/AttributesRepackStep.cs]: And it is a choice to secure the space, but why not generalize the fix and be sure more than a space or a tabulation are just also. It is a case so particular.

**Pull request closed without merging**